DailyReportWeb/App_Start/RouteConfig.cs

```diff
@@ -45,16 +45,18 @@ public static void RegisterRoutes(this RouteCollection routes)
 
         public static HttpConfiguration InitApiRoutes(this HttpConfiguration config)
         {
-			config.Routes.MapHttpRoute(
-				name: "Account",
-				routeTemplate: "api/account/{action}",
-				defaults: new { controller = "Account" }
-			);
+            config.Routes.MapHttpRoute(
+                name: "ActionApi",
+                routeTemplate: "api/{controller}/{action}",
+                defaults: new { },
+                constraints: new { action = @"^[a-zA-Z]+$" }
+            );
 
             config.Routes.MapHttpRoute(
                 name: "DefaultApi",
                 routeTemplate: "api/{controller}/{id}",
-                defaults: new { id = RouteParameter.Optional }
+                defaults: new { id = RouteParameter.Optional },
+                constraints: new { id = @"^$|\d+"}
```

Comment:
Not sure this is the best solution. The id is not necessarily long. There can be string id's as well in the future.

Reply:
I came across this problem, but I found no other way to make it generic. If the id should be a string, then the method should be named exactly and the parameter received with FromUri.

Reply:
Where does this problem occur? So that I can have a look at the controller

Reply:
App.js, Timezone. I changed in this pull request as well.

Reply:
The instances controller works the same way. There is one method with Get() and one with Get(id). And they worked both. 
The mistake that you do is that you pass the name of the method and it is considered to be the id. If you want the get method you do: http.get(api/controller) -> this gets the list. http.get(api/controller/id) -> gets the element. Try this way without the routes changed. It should work.

Reply:
Ok, and what happens if I want multiple parameters? Also, there is one hardcoded route for the account controller, so making a generic change for the difference between the id and action seems like a good thing to do.